### PR TITLE
[EC-142] Change Account to use UInt256

### PR DESCRIPTION
**Pull request by adamsmo**

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala [hunk 1]
```diff
@@ -129,7 +129,7 @@ class CallOpcodesSpec extends WordSpec with Matchers {
       "pass correct addresses and value" in {
         Address(call.extStorage.load(fxt.ownerOffset)) shouldEqual fxt.extAddr
         Address(call.extStorage.load(fxt.callerOffset)) shouldEqual fxt.ownerAddr
-        call.extStorage.load(fxt.valueOffset) shouldEqual call.value
+        call.extStorage.load(fxt.valueOffset) shouldEqual DataWord(call.value.n)
```
Comment (adamsmo) [inline src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala]: @rtkaczyk should we change DataWord to IInt256 ?

Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala]: Wait, what??? `DataWord` has been removed some time ago.

Comment (adamsmo) [inline src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala]: ok, will merge from `phase/3/txExecution` and check if it is still there

File: src/main/scala/io/iohk/ethereum/vm/ProgramState.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ case class ProgramState[W <: WorldStateProxy[W, S], S <: Storage[S]](
 
   def ownAddress: Address = env.ownerAddr
 
-  def ownBalance: BigInt = world.getBalance(ownAddress)
+  def ownBalance: UInt256 = world.getBalance(ownAddress)
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/vm/ProgramState.scala]: This has been changed before! It looks like there was a merge gone wrong

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/vm/ProgramState.scala]: or I did not updated tx execution before starting this ticket

Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/vm/ProgramState.scala]: yeah, looks better now

File: src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala [hunk 3]
```diff
@@ -82,7 +82,7 @@ class CallOpcodesSpec extends WordSpec with Matchers {
     inputData: ByteString = fxt.inputData,
     gas: UInt256 = fxt.requiredGas + fxt.gasMargin,
     to: Address = fxt.extAddr,
-    value: UInt256 = fxt.initialBalance / 2
+    value: UInt256 = fxt.initialBalance div UInt256(2)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala]: Why change this?

File: src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala [hunk 4]
```diff
@@ -100,15 +100,15 @@ class CallOpcodesSpec extends WordSpec with Matchers {
     private val stack = Stack.empty().push(if (op == DELEGATECALL) params.take(4) ++ params.drop(5) else params)
     private val mem = Memory.empty.store(UInt256.Zero, inputData)
 
-    val stateIn = ProgramState(context).withStack(stack).withMemory(mem)
-    val stateOut = op.execute(stateIn)
-    val world = stateOut.world
+    val stateIn: ProgramState[MockWorldState, MockStorage] = ProgramState(context).withStack(stack).withMemory(mem)
+    val stateOut: ProgramState[MockWorldState, MockStorage] = op.execute(stateIn)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/vm/CallOpcodesSpec.scala]: This is just for tests, honestly I like it better without explicit types. But if you have to, you can use [this](https://github.com/input-output-hk/etc-client/blob/phase/3/txExecution/src/test/scala/io/iohk/ethereum/vm/MockWorldState.scala#L8) :)

File: src/main/scala/io/iohk/ethereum/vmrunner/State.scala [hunk 5]
```diff
@@ -10,7 +10,7 @@ object State {
   val creatorAddress = Address(0xabcdef) //scalastyle:off magic.number
 
   private var currentWorld = {
-    val creatorAcc = Account.Empty.updateBalance(BigInt(10).pow(9))
+    val creatorAcc = Account.Empty.updateBalance(UInt256(10) ** UInt256(9))
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/vmrunner/State.scala]: won't `10 ** 9` work?

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/vmrunner/State.scala]: `value ** is not a member of Int`

File: src/main/scala/io/iohk/ethereum/rlp/RLPImplicitConversions.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package io.iohk.ethereum.rlp
+
+import akka.util.ByteString
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.vm.UInt256
+
+import scala.language.implicitConversions
+
+object RLPImplicitConversions {
```
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/rlp/RLPImplicitConversions.scala]: do we need both RLPImplicitConversions and RLPImplicits?

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/rlp/RLPImplicitConversions.scala]: Yes, If I put it in one file scalastyle check fails because of too many methods

**Pull request merged**